src/databricks/labs/ucx/aws/access.py

```diff
@@ -49,33 +50,37 @@ def __init__(
         self._kms_key = kms_key
         self._filename = self.INSTANCE_PROFILES_FILE_NAMES
 
-    def create_uc_roles_cli(self, *, single_role=True, role_name="UC_ROLE", policy_name="UC_POLICY"):
+    def uc_roles_list(self, *, single_role=True, role_name="UC_ROLE", policy_name="UC_POLICY"):
         # Get the missing paths
         # Identify the S3 prefixes
         # Create the roles and policies for the missing S3 prefixes
         # If single_role is True, create a single role and policy for all the missing S3 prefixes
         # If single_role is False, create a role and policy for each missing S3 prefix
-        missing_paths = self._identify_missing_paths()
+        roles: list[AWSUCRoleCandidate] = []
+        missing_paths = self.identify_missing_paths()
         s3_prefixes = set()
         for missing_path in missing_paths:
             match = re.match(AWSResources.S3_PATH_REGEX, missing_path)
             if match:
                 s3_prefixes.add(missing_path)
         if single_role:
-            if self._aws_resources.create_uc_role(role_name):
+            roles.append(AWSUCRoleCandidate(role_name, policy_name, list(s3_prefixes)))
+        else:
+            role_id = 1
+            for s3_prefix in sorted(list(s3_prefixes)):
```

Comment:
you can use `for idx, s3_prefix in enumerate(sorted(list(s3_prefixes)))`

Reply:
Good idea